pkg/skbn/abs.go

```diff
@@ -0,0 +1,184 @@
+package skbn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+var err error
+
+func validateAbsPath(pathSplit []string) error {
+	if len(pathSplit) >= 1 {
+		return nil
+	}
+	return fmt.Errorf("illegal path: %s", filepath.Join(pathSplit...))
+}
+
+func initAbsVariables(split []string) (string, string, string) {
+	account := split[0]
+	container := split[1]
+	path := filepath.Join(split[2:]...)
+
+	return account, container, path
+}
+
+func getNewPipeline() (pipeline.Pipeline, error) {
+	accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
```

Comment:
Is this the only way to authenticate to abs? Or does abs provide an implicit "credentials chain"?
For example, if these vars are defined, use them. Otherwise, do something else...
In addition, what happens if this code is to be run inside a Kubernetes pod, and you don`t want to provide these credentials, but rather use a Role? (not sure how the authentication model works in Azure).

Reply:
I've chosen this way because the actual storage may be in the another subscription that Kubernetes cluster has no access to. Second reason: az cli requires a tricky way of authentication command line apps (you have to get a token from app and go to the microsoft website to register it). I haven't heard about 1 to 1 mapping of kubernetes roles to azure roles. One more way is to use the aad authentication (clientId, clientSecret), but this is much more than this up requires.